decode_test.go

```diff
@@ -123,6 +123,21 @@ func TestExtension(t *testing.T) {
 	require.Equal(t, "hi", proto.GetExtension(actual, testv1.E_P2TStringExt))
 }
 
+func TestDiscardUnknown(t *testing.T) {
+	t.Parallel()
+
+	data := []byte(`unknown: hi`)
```

Comment:
Maybe include a few known and a few unknown fields and then verify afterwards that all of the known fields were correctly set?